src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1204,6 +1253,42 @@ async static (state) =>
             return (_success, allGraphs, allRuntimes);
         }
 
+        private bool ShouldPrunePackage(TargetFrameworkInformation projectTargetFramework, FindLibraryEntryResult refItemResult, LibraryDependency dep, bool isPackage, bool isDirectPackageReferenceFromRootProject)
+        {
+            if (projectTargetFramework!.PackagesToPrune.TryGetValue(dep.Name, out PrunePackageReference? prunableVersion))
+            {
+                if (dep.LibraryRange!.VersionRange!.Satisfies(prunableVersion.VersionRange!.MaxVersion!))
+                {
+                    if (!isPackage)
+                    {
+                        if (SdkAnalysisLevelMinimums.IsEnabled(
```

Comment:
I think this whole feature should be off by default but when enabled, we should consider a mode like "strict" where these errors are logged.  I'm thinking of the scenario someday when I update to the latest .NET SDK and now I can't restore.  If we make this on by default but just silently ignore misconfigurations, that would be a fine experience.  If I want to opt into getting these errors, that would at least be a change I made.  I think good documentation would help users know that they can opt into the more strict mode.

Reply:
Discussed briefly offline. 

I'll add this to the backlog for PrunePackageReference and work on this as we progress towards the official roll-out: 
https://github.com/NuGet/Client.Engineering/issues/3111